server/http/helloapp/f_survey.go

```diff
@@ -0,0 +1,54 @@
+package helloapp
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+func (h *helloapp) newSurveyForm(message string) *api.Form {
+	return &api.Form{
+		Title:         "Emotional response survey",
+		Header:        message,
+		Footer:        "Let the world know!",
+		SubmitButtons: fieldResponse,
+		Fields: []*api.Field{
+			{
+				Name: fieldResponse,
+				Type: api.FieldTypeStaticSelect,
+				SelectStaticOptions: []api.SelectOption{
+					{Label: "Like", Value: "like"},
+					{Label: "Dislike", Value: "dislike"},
+				},
+			},
+		},
+	}
+}
+
+func (h *helloapp) fSurvey(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, c *api.Call) (int, error) {
+	var out *api.CallResponse
+
+	// userID := c.GetValue(fieldUserID, c.Context.ActingUserID)
+	message := c.GetValue(fieldMessage, "default hello message")
+
+	switch c.Type {
+	case api.CallTypeForm:
+		out = h.newSurveyFormResponse(message)
+
+	case api.CallTypeSubmit:
+		// TODO post something somewhere; for embedded form - what do we do?
+		out = &api.CallResponse{}
```

Comment:
Per your comment here, maybe need a way to clean up / close an embedded form if appropriate?

Reply:
There's a ticket for @aaronrothschild and @abhijit-singh to define the embedded form submit behavior. 2/5 embedded forms should be immutable *including no support for `RefreshOnChange`, and collapse upon a successful submission. Results should be displayed as a separate message.